### DISCUSSION TITLE
How to insert an item at an index in a List state?

### DISCUSSION THREAD

**Discussion by anmol-me** [rrousselGit/riverpod#1866] [Q&A] [answered]
state => List list = [1, 2, 4];

Inside a state notifier, if I want to add an item to the list state, I would-

```
  void addInt(int newNumber) {
    state = [...state, newNumber];
  }
```

But now how do I add a new number '3' at the index of 2 and move the item at the index of 2 to the index of 3?

**Comment by rrousselGit** [comment, marked as answer]
```dart
state = [...state]..insert(index, value);
```

**Reply by anmol-me** [reply]
"state = [...state]..insert()"

Why does the list need cascade notation to access insert method?

**Reply by anmol-me** [reply]
The cascade operator (..) allows you to perform multiple operations on the same object without having to reference the object again.
It essentially says, "Take the object that was just created and call this method on it."
